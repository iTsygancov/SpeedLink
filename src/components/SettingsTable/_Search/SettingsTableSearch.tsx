import { Button } from "@/components/ui/button";
import { Input } from "@/components/ui/input";
import { Search } from "lucide-react";

export type SettingsTableSearchProps = {
  handleSearch: (value: string) => void;
  searchValue: string;
};

const SettingsTableSearch = ({
  handleSearch,
  searchValue
}: SettingsTableSearchProps) => {
  const handleClearSearch = () => {
    handleSearch("");
  };

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    handleSearch(e.target.value);
  };

  return (
    <div className='mb-4 flex items-center justify-center gap-4'>
      <div className='relative'>
        <Search
          className='absolute left-4 top-1/2 -translate-y-1/2 text-primary'
          size={16}
        />
        <Input
          className='w-[350px] pl-10'
          value={searchValue}
          onChange={handleInputChange}
          placeholder='Search'
        />
      </div>
      <Button variant='secondary' onClick={handleClearSearch}>
        Clear
      </Button>
    </div>
  );
};

export default SettingsTableSearch;
