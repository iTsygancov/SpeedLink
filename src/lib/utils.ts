import { type ClassValue, clsx } from "clsx";
import { twMerge } from "tailwind-merge";

export function cn(...inputs: ClassValue[]) {
  return twMerge(clsx(inputs));
}

export function generateAlphanumericKeysArray() {
  const keysArray = [];

  // Generate lowercase alphabetic keys
  for (let i = 65; i <= 90; i++) {
    keysArray.push(String.fromCharCode(i));
  }

  // Generate numeric keys
  for (let i = 48; i <= 57; i++) {
    keysArray.push(String.fromCharCode(i));
  }

  return keysArray;
}
