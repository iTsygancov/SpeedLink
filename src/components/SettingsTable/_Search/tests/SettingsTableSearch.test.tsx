import SettingsTableSearch, {
  SettingsTableSearchProps
} from "../SettingsTableSearch";
import { expectToMatchSnapshot, renderTree } from "@/test/utils";
import { describe, it, vi } from "vitest";

const initialProps: SettingsTableSearchProps = {
  handleSearch: vi.fn(),
  searchValue: ""
};

const testComponentWithProps = (props: SettingsTableSearchProps) => {
  expectToMatchSnapshot(renderTree(<SettingsTableSearch {...props} />));
};

describe("SettingsTableSearch", () => {
  it("should render with default props", () => {
    testComponentWithProps(initialProps);
  });

  it("should render with test input value", () => {
    testComponentWithProps({
      ...initialProps,
      searchValue: "test"
    });
  });
});
