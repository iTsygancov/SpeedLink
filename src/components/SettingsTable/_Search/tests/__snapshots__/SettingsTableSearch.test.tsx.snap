// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SettingsTableSearch > should render with default props 1`] = `
<div
  className="mb-4 flex items-center justify-center gap-4"
>
  <div
    className="relative"
  >
    <svg
      className="absolute left-4 top-1/2 -translate-y-1/2 text-primary"
      fill="none"
      height={16}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      viewBox="0 0 24 24"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="11"
        cy="11"
        r="8"
      />
      <path
        d="m21 21-4.3-4.3"
      />
    </svg>
    <input
      className="flex h-10 rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 w-[350px] pl-10"
      onChange={[Function]}
      placeholder="Search"
      value=""
    />
  </div>
  <button
    className="active:shadow-inner hover:shadow-sm inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2"
    onClick={[Function]}
  >
    Clear
  </button>
</div>
`;

exports[`SettingsTableSearch > should render with test input value 1`] = `
<div
  className="mb-4 flex items-center justify-center gap-4"
>
  <div
    className="relative"
  >
    <svg
      className="absolute left-4 top-1/2 -translate-y-1/2 text-primary"
      fill="none"
      height={16}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      viewBox="0 0 24 24"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="11"
        cy="11"
        r="8"
      />
      <path
        d="m21 21-4.3-4.3"
      />
    </svg>
    <input
      className="flex h-10 rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 w-[350px] pl-10"
      onChange={[Function]}
      placeholder="Search"
      value="test"
    />
  </div>
  <button
    className="active:shadow-inner hover:shadow-sm inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2"
    onClick={[Function]}
  >
    Clear
  </button>
</div>
`;
